src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const object of actions) {
+    for (const key in object) {
+      switch (object[key]) {
+        case 'addProperties':
+          for (const iKey in object.extraData) {
+            state[iKey] = object.extraData[iKey];
+          }
+          break;
+
+        case 'removeProperties':
+          for (const value of object.keysToRemove) {
+            if (state[value]) {
+              delete state[value];
+            }
+          }
+          break;
+
+        case 'clear':
+          for (const iKey in state) {
+            delete state[iKey];
+          }
+          break;
+      }
```

Comment:
`switch/case` should always have `default` case for error handling. Make it return string `Error`.